builtin/credential/aws-ec2/path_login.go

```diff
@@ -396,6 +412,11 @@ func (b *backend) pathLoginUpdate(
 				"role_tag_max_ttl": rTagMaxTTL.String(),
 				"role":             roleName,
 				"ami_id":           identityDoc.AmiID,
+				// Echo the client nonce back. If nonce was not
+				// supplied to the endpoint, callers should
+				// extract out the nonce from this field for
+				// reauthentication requests.
+				"nonce": clientNonce,
```

Comment:
Can we make this condition on authentication actually being allowed? So if it isn't, don't return a nonce.